### ISSUE TITLE
[openstack] inconsistent model-collection methods to get by id

### ISSUE THREAD

**Issue by dhague** [fog/fog#3726]
In Fog, the model collections classes (typically the pluralised names in lib/fog/*provider*/models/*service*) have a get(id) method to fetch a model object by its ID.
In the case where the ID cannot be found, the get(id) method typically swallows the NotFound exception and returns nil.

In the OpenStack provider case, some classes use get(id) as described above, some use get(id) but do not swallow the NotFound exception, some classes use find_by_id(id) and not not swallow the exception, and in one case both methods are supported with find_by_id being marked as deprecated. 

In other words, there is a great deal of inconsistency in how to get a model object by its ID in the Fog OpenStack provider.

**Comment by dhague** [comment]
I would propose to resolve this problem as follows (Note: this will introduce some breaking changes, but as we are moving from 1.x to 2.0 I think there is no better time in which to do so):

* Standardize on implementing get(id), swallowing any NotFound exception and returning nil in that case (this is the breaking change for those implementations of get(id) which do not currently swallow the exception)
* Leave find_by_id(id) methods in place, but mark them as deprecated using the same style as in Fog::Orchestration::OpenStack::Stacks (https://github.com/fog/fog/blob/v1.34.0/lib/fog/openstack/models/orchestration/stacks.rb#L24)

Thoughts?

**Comment by Ladas** [comment]
@dhague I 've started the standardization work here https://github.com/fog/fog/blob/v1.34.0/lib/fog/openstack/models/collection.rb#L37

we can alias the method with deprecation on one place

I think I am fine with the breaking change.

**Comment by dhague** [comment]
What about the difference in behaviour, namely **get** swallowing the NotFound exception but **find_by_id** passing it through? It seems that using alias will unnecessarily break that behaviour.

**Comment by geemus** [comment]
Overall sounds good. I would lean toward swallowing the error (more consistent with other parts of fog and I think a nice default), but it is breaking.

**Comment by Ladas** [comment]
@dhague sorry I misread the part, I was on my way out in the evening :-)

seems like there is always one common 'get' behavior. What about making private method '_get' on each class, that would fetch the record. And on the super class we would have public method 'get' that will catch exceptions of '_get'. And 'find_by_id' that will just log deprecation and call '_get'? The exception will always be the same NotFound, right? So the logic could be on one place.

Also I am thinking about how to design the method when the resource is nested, I think I saw max 2 keys used for identification, e.g. here https://github.com/fog/fog/blob/v1.34.0/lib/fog/openstack/models/orchestration/resources.rb#L19

What do you think?

**Comment by dhague** [comment]
@Ladas sounds like a good plan - I'll get to work on it.

**Comment by Ladas** [comment]
@dhague awesome, if you will need any help, let me know :-)

**Comment by plribeiro3000** [comment]
Closing in favor of fog/fog-openstack#52
